barretenberg/cpp/src/barretenberg/proof_system/relations/non_native_field_relation.hpp

```diff
@@ -0,0 +1,286 @@
+        const auto& v_2_0 = relation_parameters.batching_challenge_v[2][0];
+        const auto& v_2_1 = relation_parameters.batching_challenge_v[2][1];
+        const auto& v_2_2 = relation_parameters.batching_challenge_v[2][2];
+        const auto& v_2_3 = relation_parameters.batching_challenge_v[2][3];
+        const auto& v_2_4 = relation_parameters.batching_challenge_v[2][4];
+        const auto& v_3_0 = relation_parameters.batching_challenge_v[3][0];
+        const auto& v_3_1 = relation_parameters.batching_challenge_v[3][1];
+        const auto& v_3_2 = relation_parameters.batching_challenge_v[3][2];
+        const auto& v_3_3 = relation_parameters.batching_challenge_v[3][3];
+        const auto& v_3_4 = relation_parameters.batching_challenge_v[3][4];
+
+        const auto& op = View(in.op);
+        const auto& p_x_low_limbs = View(in.p_x_low_limbs);
+        const auto& p_y_low_limbs = View(in.p_y_low_limbs);
+        const auto& p_x_high_limbs = View(in.p_x_high_limbs);
+        const auto& p_y_high_limbs = View(in.p_y_high_limbs);
+        const auto& accumulators_binary_limbs_0 = View(in.accumulators_binary_limbs_0);
+        const auto& accumulators_binary_limbs_1 = View(in.accumulators_binary_limbs_1);
+        const auto& accumulators_binary_limbs_2 = View(in.accumulators_binary_limbs_2);
+        const auto& accumulators_binary_limbs_3 = View(in.accumulators_binary_limbs_3);
+        const auto& z_low_limbs = View(in.z_low_limbs);
+        const auto& z_high_limbs = View(in.z_high_limbs);
+        const auto& quotient_low_binary_limbs = View(in.quotient_low_binary_limbs);
+        const auto& quotient_high_binary_limbs = View(in.quotient_high_binary_limbs);
+        const auto& p_x_low_limbs_shift = View(in.p_x_low_limbs_shift);
+        const auto& p_y_low_limbs_shift = View(in.p_y_low_limbs_shift);
+        const auto& p_x_high_limbs_shift = View(in.p_x_high_limbs_shift);
+        const auto& p_y_high_limbs_shift = View(in.p_y_high_limbs_shift);
+        const auto& accumulators_binary_limbs_0_shift = View(in.accumulators_binary_limbs_0_shift);
+        const auto& accumulators_binary_limbs_1_shift = View(in.accumulators_binary_limbs_1_shift);
+        const auto& accumulators_binary_limbs_2_shift = View(in.accumulators_binary_limbs_2_shift);
+        const auto& accumulators_binary_limbs_3_shift = View(in.accumulators_binary_limbs_3_shift);
+        const auto& z_low_limbs_shift = View(in.z_low_limbs_shift);
+        const auto& z_high_limbs_shift = View(in.z_high_limbs_shift);
+        const auto& quotient_low_binary_limbs_shift = View(in.quotient_low_binary_limbs_shift);
+        const auto& quotient_high_binary_limbs_shift = View(in.quotient_high_binary_limbs_shift);
+        const auto& relation_wide_limbs = View(in.relation_wide_limbs);
+        const auto& relation_wide_limbs_shift = View(in.relation_wide_limbs_shift);
+        const auto& lagrange_odd = View(in.lagrange_odd);
```

Comment:
Can you explain in the block comment how this Lagrange polynomial is to be used?

Reply:
Added